src/charm.py

```diff
@@ -169,6 +170,15 @@ def config_containerd_registries(self, _: ConfigChangedEvent):
                 "failed to apply containerd_custom_registries, check logs for details"
             )
 
+    def config_rbac(self, _: ConfigChangedEvent):
+        if isinstance(self.unit.status, BlockedStatus):
+            return
+
+        if self._state.joined:
+            LOG.info("Ensure RBAC mode is %s", self.config["rbac"])
+            microk8s.wait_ready()
+            microk8s.configure_rbac(self.config["rbac"])
+
```

Comment:
Please set the unit status to `Maintenance` before starting to configure RBAC, see for example in `config_extra_sans`